compiler/damlc/tests/daml-test-files/OverAppliedFoldEvalOrder.daml

```diff
@@ -0,0 +1,12 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- @ERROR Aborted: BOOM
+
+module OverAppliedFoldEvalOrder where
+
+boom = scenario do
+  let id x = x
+  let boomCompose _ _ = error "BOOM"
+  let _ = foldr boomCompose id [id] (error "BANG")
```

Comment:
It's funny how my "BOOM" and "BANG" made into here. Maybe it's better to follow @sofiafaro-da's style and call the errors "OK" and "Failed" (or something like "Expected" and "Unexpected" if you prefer that) in order to make it clearer which error is the one that should happen and which one is the one we don't want.

Reply:
This test predated Sofia's. I updated to her style.